pkg/registry/csv.go

```diff
@@ -53,6 +62,50 @@ type ClusterServiceVersion struct {
 	Spec json.RawMessage `json:"spec"`
 }
 
+// ReadCSVFromBundleDirectory tries to parse every YAML file in the directory and see if they are CSV.
+// According to the strict one CSV rule for every bundle, we return the first file that is considered a CSV type.
+func ReadCSVFromBundleDirectory(bundleDir string) (*ClusterServiceVersion, error) {
+	files, err := ioutil.ReadDir(bundleDir)
+	if err != nil {
+		return nil, fmt.Errorf("error reading bundle directory %s, %v", bundleDir, err)
+	}
+
+	ymlFiles := []string{}
+	for _, f := range files {
+		if !f.IsDir() {
+			ymlFiles = append(ymlFiles, f.Name())
```

Comment:
shouldn't we use the yaml decoder here to see if the file is actually a yaml file?

Reply:
This just grabs all files. I will rename the variables. This gets all files, then it unmarshals to yaml. then it tries to see if it is csv.